packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1750,21 +1768,22 @@ Run 'flutter -h' (or 'flutter <command> -h') for available flutter commands and
         usage: globals.flutterUsage,
         analytics: analytics,
         projectDir: project.directory,
+        packageConfigPath: packageConfigPath(),
         generateDartPluginRegistry: true,
       );
 
-      await generateLocalizationsSyntheticPackage(
-        environment: environment,
-        buildSystem: globals.buildSystem,
-        buildTargets: globals.buildTargets,
-      );
-
       await pub.get(
         context: PubContext.getVerifyContext(name),
         project: project,
         checkUpToDate: cachePubGet,
       );
 
+      await generateLocalizationsSyntheticPackage(
```

Comment:
why did you move this after the pub.get? I would think that we would want to pub get after the synthetic package is created?

Reply:
It depends on `environment` that now contains the package config path. That is not valid before pub get has run.
Also I don't see what pub has to use the synthetic package for...

Reply:
i trust you :)

Reply:
I really don't have a good feel for the generateLocalizationsSyntheticPackage mechanism. But I don't believe it interacts with package resolution besides creating a package that is injected into package_config.json. So fingers crossed the order should not really matter.

Reply:
My naive understanding was that the project's package_config.json would need to have a reference to the synthetic package, and thus it would need to exist so that pub would find it and write it to package_config.json.

@jonahwilliams we might need your help here...

Reply:
pub itself will never write the synthetic package into package config. That is done by flutter_tool here:
https://github.com/flutter/flutter/blob/aaab2e145f20c87c27b39b7c4a382733a420e8fc/packages/flutter_tools/lib/src/dart/pub.dart#L751

That happens at the end of `flutter pub get`: https://github.com/flutter/flutter/blob/aaab2e145f20c87c27b39b7c4a382733a420e8fc/packages/flutter_tools/lib/src/dart/pub.dart#L369

Reply:
ohhhh, yeah, I forgot about that. :)